src/Language/Haskell/TH/Datatype.hs

```diff
@@ -202,6 +203,11 @@ data DatatypeInfo = DatatypeInfo
   , datatypeVars      :: [TyVarBndrUnit]   -- ^ Type parameters
   , datatypeInstTypes :: [Type]            -- ^ Argument types
   , datatypeVariant   :: DatatypeVariant   -- ^ Extra information
+  , datatypeReturnKind:: Kind              -- ^ Return Kind of the type
+                                           --
+                                           -- Evaluates to an error before GHC-8.10/template-haskell-2.16.
```

Comment:
In the newest version of this PR, it is no longer an error to use `datatypeReturnKind` before GHC 8.10, as `th-abstraction` will simply default to `StarT`. Is that what you intended? If so, then please update these Haddocks accordingly.

Reply:
Yes good point! I'll update it.
I think it makes sense to always default to StarT since then we have something consistent between:
- failure because we can't get a tycon's kind
- failure because we are on too old a version of GHC